specification/cognitiveservices/OpenAI.Inference/models/batch.tsp

```diff
@@ -0,0 +1,269 @@
+  @encodedName("application/json", "expired_at")
+  @encode("unixTimestamp", int32)
+  expiredAt?: utcDateTime;
+
+  /** The Unix timestamp (in seconds) for when the batch started cancelling. */
+  @encodedName("application/json", "cancelling_at")
+  @encode("unixTimestamp", int32)
+  cancellingAt?: utcDateTime;
+
+  /** The Unix timestamp (in seconds) for when the batch was cancelled. */
+  @encodedName("application/json", "cancelled_at")
+  @encode("unixTimestamp", int32)
+  cancelledAt?: utcDateTime;
+
+  /** The request counts for different statuses within the batch. */
+  @encodedName("application/json", "request_counts")
+  requestCounts?: {
+    /** Total number of requests in the batch. */
+    total: int32;
+
+    /** Number of requests that have been completed successfully. */
+    completed: int32;
+
+    /** Number of requests that have failed. */
+    failed: int32;
+  };
+
+  /** A set of key-value pairs that can be attached to the batch. This can be useful for storing additional information about the batch in a structured format. */
+  metadata?: Record<string>;
+}
+
+/** The per-line object of the batch input file */
+@added(ServiceApiVersions.v2024_07_01_Preview)
+model BatchRequestInput {
+  /** A developer-provided per-request id that will be used to match outputs to inputs. Must be unique for each request in a batch. */
+  @encodedName("application/json", "custom_id")
+  customId?: string;
+
+  @doc("""
+    The HTTP method to be used for the request. Currently only `POST` is supported.
+    """)
+  method?: "POST";
+
+  @doc("""
+    The OpenAI API relative URL to be used for the request. Currently `/v1/chat/completions`, `/v1/embeddings`, and `/v1/completions` are supported.
+    """)
+  url?: url;
+}
+
+/** The per-line object of the batch output and error files */
+@added(ServiceApiVersions.v2024_07_01_Preview)
+model BatchRequestOutput {
+  /** The Id of the request. */
+  id?: string;
+
+  /** A developer-provided per-request id that will be used to match outputs to inputs. */
+  @encodedName("application/json", "custom_id")
+  customId?: string;
+
+  /** The http response */
+  response?: {
+    /** The HTTP status code of the response */
+    @encodedName("application/json", "status_code")
+    statusCode?: int32;
+
+    /** An unique identifier for the OpenAI API request. Please include this request ID when contacting support. */
+    @encodedName("application/json", "request_id")
+    requestId?: string;
+
+    /** The JSON body of the response */
+    body?: Record<string>;
+  };
+
+  /** For requests that failed with a non-HTTP error, this will contain more information on the cause of the failure. */
+  error?: {
+    /** A machine-readable error code. */
+    code?: string;
+
+    /** A human-readable error message. */
+    message?: string;
+  };
+}
+
+/** A list of paginated Batch objects. */
+@added(ServiceApiVersions.v2024_07_01_Preview)
+model ListBatchesResponse {
```

Comment:
Personally, I think that we should lift the `common/models.tsp` definition of `OpenAIPageableListOf<T>` . I will push a commit adding this (and feel free to revert), but just wondering if this was on purpose or a tiny slip.